tests/components/fyta/test_config_flow.py

```diff
@@ -0,0 +1,190 @@
+        return_value=AsyncMock(),
+    ) as mock:
+        fyta = mock.return_value
+        fyta.login.side_effect = FytaConnectionError
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+        )
+        await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    # tests with authentication error
+    with patch(
+        "homeassistant.components.fyta.config_flow.FytaConnector",
+        return_value=AsyncMock(),
+    ) as mock:
+        fyta = mock.return_value
+        fyta.login.side_effect = FytaAuthentificationError
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+        )
+        await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "auth_error"}
+
+    # tests with password error
+    with patch(
+        "homeassistant.components.fyta.config_flow.FytaConnector",
+        return_value=AsyncMock(),
+    ) as mock:
+        fyta = mock.return_value
+        fyta.login.side_effect = FytaPasswordError
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+        )
+        await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "auth_error", CONF_PASSWORD: "password_error"}
+
+    # tests with other error
+    with patch(
+        "homeassistant.components.fyta.config_flow.FytaConnector",
+        return_value=AsyncMock(),
+    ) as mock:
+        fyta = mock.return_value
+        fyta.login.side_effect = Exception
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+        )
+        await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "unknown"}
+
+    # tests with all information provided
+    with patch(
+        "homeassistant.components.fyta.config_flow.FytaConnector",
+        return_value=AsyncMock(),
+    ) as mock:
+        fyta = mock.return_value
+        fyta.login.return_value = {
+            "access_token": ACCESS_TOKEN,
+            "expiration": EXPIRATION,
+        }
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD}
+        )
+        await hass.async_block_till_done()
```

Comment:
Please use parametrize for this. I think either `epion`, `analytics_insights` or `flexit_bacnet` has it